packages/backend/src/services/ProjectService/ProjectService.ts

```diff
@@ -938,13 +938,29 @@ export class ProjectService {
             explore,
         });
 
+        const suggestionsToAddToFields = dashboardFilters
+            ? metricQuery.dimensions.reduce((sum, dimensionId) => {
+                  //! this is bad, but necessary for UX - optimise so it's not extremely slow when there are many rows
+                  const newSuggestions: string[] =
+                      rows.reduce<string[]>((acc, row) => {
+                          const value = row[dimensionId]?.value.raw;
+                          if (typeof value === 'string') {
+                              return [...acc, value];
```

Comment:
I'm not sure how big these get, but if we are concerned about perf I think concat will be a little faster than the spread.

Reply:
💯 , I've removed this code to see if we _actually_ need it. But your suggestion makes a lot of sense 👍